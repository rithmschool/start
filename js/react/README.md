<h1 align="center">
  Rithm React Starter
</h1>

  A mildly-customized fork of <a href="https://github.com/SafdarJamal/vite-template-react">Vite Template React</a>.

<p align="center">
  <a href="https://github.com/SafdarJamal/vite-template-react/releases">
    <img src="https://img.shields.io/github/v/release/SafdarJamal/vite-template-react" alt="GitHub Release (latest by date)" />
  </a>
  <a href="https://github.com/SafdarJamal/vite-template-react/blob/main/LICENSE">
    <img src="https://img.shields.io/github/license/SafdarJamal/vite-template-react" alt="License" />
  </a>
</p>

<p align="center">
    A <a href="https://vitejs.dev">Vite</a> + <a href="https://react.dev">React</a> starter template.
</p>


## Folder Structure

No configuration or complicated folder structures, just the files you need to build your app:

```
vite-template-react
├── node_modules
├── public
│   ├── favicon.svg
└── src
    ├── App.css
    ├── App.jsx
    ├── App.test.jsx
    ├── index.css
    ├── index.jsx
    └── logo.svg
    └── setupTests.js
├── .gitignore
├── index.html
├── package.json
├── vite.config.js
```

## Development

To get a local copy of the code, use degit:

```
npm install -g degit    # only need to do this once, ever

degit rithmschool/react-starter my-project-name
cd my-project-name
```

You can start a local web server by running:

```
npm start
```

And then open http://localhost:3000 to view it in the browser.

#### Available Scripts

In this project, you can run the following scripts:

| Script        | Description                                             |
| ------------- | ------------------------------------------------------- |
| npm start     | Runs the app in the development mode.                   |
| npm test      | Launches the test runner in the interactive watch mode. |
| npm run build | Builds the app for production to the `dist` folder.     |

## Credits

Customizations by Joel Burton (joel@rithmschool.com), based on the
Vite Template React, which is built and maintained by [Safdar Jamal](https://safdarjamal.github.io).

## License

This project is licensed under the terms of the [MIT license](https://github.com/SafdarJamal/vite-template-react/blob/main/LICENSE).
