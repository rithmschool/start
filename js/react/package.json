{
  "name": "rithmschool-react-starter",
  "version": "1.4.0",
  "type": "module",
  "scripts": {
    "start": "vite --port 3000 --open",
    "dev": "vite --port 3000 --open",
    "build": "vite build",
    "test": "vitest",
    "coverage": "vitest --coverage",
    "test:debug": "vitest --inspect-brk"
  },
  "dependencies": {
    "bootstrap": "^5.3.2",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@testing-library/jest-dom": "^6.1.6",
    "@testing-library/react": "^14.1.2",
    "@vitejs/plugin-react": "^4.2.1",
    "@vitest/coverage-v8": "^1.2.2",
    "jsdom": "^23.0.1",
    "vite": "^5.0.8",
    "vitest": "^1.1.1"
  }
}
