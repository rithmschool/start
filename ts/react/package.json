{
  "name": "rithmschool-react-starter",
  "version": "1.4.0",
  "type": "module",
  "scripts": {
    "start": "vite --port 3000 --open",
    "dev": "vite --port 3000 --open",
    "build": "vite build",
    "test": "vitest"
  },
  "dependencies": {
    "bootstrap": "^5.3.2",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@testing-library/jest-dom": "^6.4.1",
    "@testing-library/react": "^14.2.1",
    "@tsconfig/vite-react": "^3.0.0",
    "@types/bootstrap": "^5.2.10",
    "@types/jsdom": "^21.1.6",
    "@types/react": "^18.2.52",
    "@types/react-dom": "^18.2.18",
    "@vitejs/plugin-react": "^4.2.1",
    "jsdom": "^23.0.1",
    "vite": "^5.0.8",
    "vitest": "^1.1.1"
  }
}
