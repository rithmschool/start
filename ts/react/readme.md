# React TS Starter

Run `npm install` and then `npm run ...`:

| Command     | Description          |
|-------------|----------------------|
| start       | Start                |
|  build      | Build to dist dir    |
| test        | Run tests            |
| test:cov    | Run tests w/coverage |
| test:debug  | Run tests w/debugger |


### Folder Structure

No configuration or complicated folder structures, just the files you need to build your app:

```
vite-template-react
├── node_modules
├── public
│   ├── favicon.svg
└── src
    ├── App.css
    ├── App.tsx
    ├── App.test.tsx
    ├── index.css
    ├── index.tsx
    └── logo.svg
    └── setupTests.ts
├── .gitignore
├── index.html
├── package.json
├── vite.config.ts
```

## Credits

Customizations by Joel Burton (joel@rithmschool.com), based on the
Vite Template React, which is built and maintained by [Safdar Jamal](https://safdarjamal.github.io).

## License

This project is licensed under the terms of the [MIT license](https://github.com/SafdarJamal/vite-template-react/blob/main/LICENSE).
