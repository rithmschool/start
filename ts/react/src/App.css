.App {
  min-height: 90vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: crimson;
  text-align: center;
}

.App-logo {
  height: 20vmin;
}

.App-link {
  color: darkgray;
}
