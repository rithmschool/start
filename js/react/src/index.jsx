import React from 'react';
import ReactDOM from 'react-dom/client';

import App from './App.jsx';

import 'bootstrap/dist/css/bootstrap.min.css';
import './index.css';


const root = ReactDOM.createRoot(document.getElementById('root'));

root.render(
  <App />
);
