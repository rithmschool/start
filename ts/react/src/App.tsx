import { useState } from "react";
import "./App.css";


/** Component for entire page.
 *
 * Props: none
 * State: none
 *
 */

function App() {
  const [count, setCount] = useState(0);

  function incrCount() {
    setCount(count => count + 1);
  }

  return (
    <div className="App">
      <main>
        <h1>Rithm React Starter</h1>
        <p>
          <button
            className="btn btn-primary"
            onClick={ incrCount }>
            Clicked: { count }
          </button>
        </p>
        <p>
          Edit <code>src/App.tsx</code>
        </p>
      </main>
    </div>
  );
};

export default App;
