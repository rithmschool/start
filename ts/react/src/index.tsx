import ReactDOM from 'react-dom/client';
import './index.css';
import App from './App.js';
import 'bootstrap/dist/css/bootstrap.min.css';

const root = ReactDOM.createRoot(document.getElementById('root')!);

root.render(
  <App />
);
