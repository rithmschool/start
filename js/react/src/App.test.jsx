import { describe, test, it, expect } from "vitest";
import { render, screen, fireEvent } from '@testing-library/react';

import App from './App.jsx';


test('renders counter button', function () {
  render(<App />);
  const buttonElement = screen.getByText(/Clicked: 0/i);
  expect(buttonElement).toBeInTheDocument();
});

test('count increases on button click', function () {
  render(<App />);
  const buttonElement = screen.getByText(/Clicked: 0/i);
  fireEvent.click(buttonElement);
  expect(buttonElement).toHaveTextContent(/Clicked: 1/i);
});

test('matches snapshot', function () {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
