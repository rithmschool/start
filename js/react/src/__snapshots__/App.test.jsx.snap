// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <main>
      <img
        alt="Rithm"
        class="App-logo"
        src="/src/logo.svg"
      />
      <h1>
        Rithm React Starter
      </h1>
      <p>
        <button
          class="btn btn-primary"
        >
          Clicked: 
          0
        </button>
      </p>
      <p>
        Edit 
        <code>
          src/App.jsx
        </code>
      </p>
    </main>
  </div>
</div>
`;
